gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -287,22 +287,75 @@ function plymeta:CanCarryWeapon(wep)
 	if not wep or not wep.Kind then
 		return false
 	end
-
+	
+	--appeareantly TTT can't handle two times the same weapon
+	for k, v in pairs(self:GetWeapons()) do
+		if IsValid(wep) and IsValid(v) and v:GetClass() == wep:GetClass() then
+			return false
+		elseif istable(wep) and IsValid(v) and v:GetClass() == wep.ClassName then
+			return false
+		end
+	end
+	
 	return self:CanCarryType(wep.Kind)
 end
 
 function plymeta:CanCarryType(t)
 	if not t then
 		return false
 	end
+	
+	return InventorySlotFree(self, t)
+end
 
-	for _, w in pairs(self:GetWeapons()) do
-		if w.Kind and w.Kind == t then
-			return false
+function plymeta:GetInventory()
+	CleanupInventoryIfDirty(self)
+	
+	return self.inventory
+end
+
+function plymeta:GetWeaponsOnSlot(slot)
+	if slot > WEAPON_EXTRA then
+		return
+	end
+	
+	CleanupInventoryIfDirty(self)
+	
+	if istable(self.inventory) then
```

Comment:
This sould always be a tablt. Just use the faster check: if self.inventory then... end